hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java

```diff
@@ -246,17 +266,22 @@ public int read(String table, String key, Set<String> fields, HashMap<String,Byt
             //do nothing for now...need to understand HBase concurrency model better
             return ServerError;
         }
-
-        if (r.isEmpty()) {
+        if (multiGetBatchSize == 0) {
+          if (r[0].isEmpty()) {
             return NoMatchingRecord;
+          }
         }
-        for (Cell c : r.listCells()) {
-            result.put(Bytes.toString(CellUtil.cloneQualifier(c)),
-                    new ByteArrayByteIterator(CellUtil.cloneValue(c)));
-            if (_debug) {
-                System.out.println("Result for field: "+Bytes.toString(CellUtil.cloneQualifier(c))+
-                        " is: "+Bytes.toString(CellUtil.cloneValue(c)));
+        for (Result res : r) {
+          if(!res.isEmpty()) {
+            for (Cell c : res.listCells()) {
+              result.put(Bytes.toString(CellUtil.cloneQualifier(c)),
```

Comment:
This seems to be common across other bindings as well.  I attributed this to the need to pull the data out of the database specific object into the java object that your app will be using.  I think every app will want to do this rather than transport the DB's proprietary object.  In that respect, it makes sense to do it and log the time in a benchmark as long as we're consistent across the bindings for each DB.